src/dataflow/components/dataflow-program.tsx

```diff
@@ -128,9 +128,15 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
       // Can this be in a control with stores injected?
       autorun(() => {
         const { hubStore } = this.stores;
-        let newChannel = false;
+        // remove any channels that are no longer active
+        this.channels = this.channels.filter(ch => {
+          const hub = hubStore.hubs.get(ch.hubId);
+          return hub && hub.hubChannels.find(hCh => hCh.id = ch.channelId);
+        });
+
```

Comment:
Not sure if it might be better to just clear the entire `channels` array and remake it each time.